protocols/virtualdefund/virtualdefund.go

```diff
@@ -366,7 +366,7 @@ func (o Objective) Update(event protocols.ObjectiveEvent) (protocols.Objective,
 
 	updated := o.clone()
 
-	for _, ss := range event.SignedStates {
+	if ss := event.SignedState; len(ss.Signatures()) != 0 {
```

Comment:
these checks guard against attempts at processing zero-valued SignStates

Reply:
This is outside the scope of this PR but should we have a canonical way of checking if a `SignedState` is a zero value? Here's another place we perform a similar [check](https://github.com/statechannels/go-nitro/blob/ag/fix-wait-for-completed-objective-ids/protocols/messages.go#L35:L35)

Reply:
I don't know if it's better to have an `IsNonZero` method or to have pointer structs that enable nil checks.